src/map/entities/battleentity.cpp

```diff
@@ -566,6 +566,12 @@ uint16 CBattleEntity::GetRangedWeaponRank()
     {
         return (weapon->getDamage() + getMod(Mod::RANGED_DMG_RANK)) / 9;
     }
+
+    if (auto* weapon = dynamic_cast<CItemWeapon*>(m_Weapons[SLOT_AMMO]))
+    {
+        return (weapon->getDamage() + getMod(Mod::RANGED_DMG_RANK)) / 9;
+    }
```

Comment:
Since it returns the same damage values is it cleaner if you use `(m_Weapons[SLOT_RANGED] ? m_Weapons[SLOT_RANGED] : m_Weapons[SLOT_AMMO])` inside the if before this?

Reply:
suppose it doesn't matter as long as there's a dynamic cast

Reply:
alright, changed it